File: split/fetch.pas
procedure fetch(a: halfword);
begin
  curc := mem[a].hh.b1;
  curf := eqtb[3935 + mem[a].hh.b0 + cursize].hh.rh;
  if curf = 0 then{723:}

  begin
    begin
      if interaction = 3 then;
      printnl(262);
      print(338);
    end;
    printsize(cursize);
    printchar(32);
    printint(mem[a].hh.b0);
    print(883);
    print(curc - 0);
    printchar(41);
    begin
      helpptr := 4;
      helpline[3] := 884;
      helpline[2] := 885;
      helpline[1] := 886;
      helpline[0] := 887;
    end;
    error;
    curi := nullcharacter;
    mem[a].hh.rh := 0;
  end{:723}

  else
  begin
    if (curc - 0 >= fontbc[curf]) and (curc - 0 <= fontec[curf]) then
      curi := fontinfo[charbase[curf] + curc].qqqq
    else
      curi := nullcharacter;
    if not ((curi.b0 > 0)) then
    begin
      charwarning(curf, curc - 0);
      mem[a].hh.rh := 0;
    end;
  end;
end;

File: cpp/fetch.cpp
#include "fetch.h"
#include "printnl.h"
#include "print.h"
#include "printsize.h"
#include "printchar.h"
#include "printint.h"
#include "error.h"
#include "charwarning.h"
#include "texte.h"

void fetch(halfword a)
{
	curc = subtype(a);
	curf = fam_fnt(type(a)+cursize);
	if (curf == 0)
	{
		printnl("! ");
		print("");
		printsize(cursize);
		printchar(' ');
		printint(type(a));
		print(" is undefined (character ");
		print(std::string(1, char(curc)));
		printchar(')');
		helpptr = 4;
		helpline[3] = txt("Somewhere in the math formula just ended, you used the");
		helpline[2] = txt("stated character from an undefined font family. For example,");
		helpline[1] = txt("plain TeX doesn't allow \\it or \\sl in subscripts. Proceed,");
		helpline[0] = txt("and I'll try to forget that I needed that character.");
		error();
		curi = nullcharacter;
		link(a) = 0;
	}
	else
	{
		if (curc >= fontbc[curf] && curc <= fontec[curf])
			curi = char_info(curf, curc);
		else
			curi = nullcharacter;
		if (skip_byte(curi) <= 0)
		{
			charwarning(curf, curc);
			link(a) = 0;
		}
	}
}
